core/runtime/common/storage_code_provider.cpp

```diff
@@ -45,9 +47,10 @@ namespace kagome::runtime {
 
       auto hash = runtime_upgrade_tracker_->getLastCodeUpdateHash(state);
       if (hash.has_value()) {
-        if (auto code_it = code_substitutes_->find(hash.value());
-            code_it != code_substitutes_->end()) {
-          OUTCOME_TRY(uncompressCodeIfNeeded(code_it->second, cached_code_));
+        if (code_substitutes_->find(hash.value()) != code_substitutes_->end()) {
```

Comment:
Minor: this way a bit shorter:
```suggestion
        if (code_substitutes_->count(hash.value())) {
```